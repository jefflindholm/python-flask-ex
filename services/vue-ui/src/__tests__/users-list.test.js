import { mount } from '@vue/test-utils';
import { createRenderer } from 'vue-server-renderer'
import UsersList from '../components/users-list';

const users = [
  { active: true, email: 'jeff@lindholm.org', id: 1, username: 'jeff' },
  { active: true, email: 'iryna@lindholm.org', id: 2, username: 'iryna' },
];

test('UsersList renders properly', () => {
  const vm = mount(UsersList, { propsData: { users }});
  const h4 = vm.findAll('h4');
  expect(h4).toHaveLength(users.length);
  expect(h4.at(0).text()).toContain('jeff');
  ['card', 'card-body', 'bg-light'].forEach(c => expect(h4.at(0).classes()).toContain(c) );
});

test('UsersList renders correct snapshot', () => {
  const renderer = createRenderer();
  const vm = mount(UsersList, { propsData: { users }});
  renderer.renderToString(vm.vm, (err, str) => {
    if (err) throw new Error(err);
    expect(str).toMatchSnapshot();
  });
})
