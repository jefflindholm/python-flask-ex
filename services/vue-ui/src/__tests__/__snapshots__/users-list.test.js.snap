// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersList renders correct snapshot 1`] = `
<div>
  <h4 class="card card-body bg-light">
    jeff
  </h4>
  <h4 class="card card-body bg-light">
    iryna
  </h4>
</div>
`;
